### PR TITLE
Add Point and Vector structs

### PR DESCRIPTION
**Pull request by jbeenie**

### CODE REVIEW

File: PathKit/Sources/PathKit/PathKit.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//// Copyright © 2019 JABT Labs. All rights reserved.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/PathKit.h]: Is the double comment start coming from the template? We should fix that.

Comment (jbeenie) [reply PathKit/Sources/PathKit/PathKit.h]: done in previous PR

File: PathKit/Sources/PathKit/Vector.swift [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    }
+
+    /// Returns the scalar division of vector by a scalar.
+    public static func / (lhs: Vector, rhs: Double) -> Vector {
+        return Vector(dx: lhs.dx / rhs, dy: lhs.dy / rhs)
+    }
+
+    /// Returns the inverse vector of `self`.
+    public static prefix func - (vector: Vector) -> Vector {
+        return -1 * vector
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Vector.swift]: I would do `Vector(dx: -dx, dy: -dy)` a lot cheaper than multiplication.

File: PathKit/Sources/PathKit/Vector.swift [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        lhs.dx *= rhs
+        lhs.dy *= rhs
+    }
+
+    public static func /= (lhs: inout Vector, rhs: Double) {
+        lhs.dx /= rhs
+        lhs.dy /= rhs
+    }
+
+    /// Returns the vector's magnitude.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Vector.swift]:
```suggestion
    /// Returns the vector's length.
```

File: PathKit/Sources/PathKit/Vector.swift [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    public var length: Double {
+        return hypot(dx, dy)
+    }
+
+    /// Returns the vector's magnitude squared.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Vector.swift]:
```suggestion
    /// Returns the square of the vector's length.
```

File: PathKit/Sources/PathKit/Point.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+// including terms governing use, modification, and redistribution, is
+// contained in the file LICENSE at the root of the source code distribution
+// tree.
+
+import Foundation
+
+/// A data structure used to represnet a point in 2-dimensional Cartesian coordinate system.
+/// The components of the point are represented using double precision floating points.
+public struct Point: Equatable {
+    /// The x-component of the point.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Point.swift]:
```suggestion
    /// The x-coordinate of the point.
```

File: PathKit/Sources/PathKit/Point.swift [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    public var x: Double
+
+    /// The y-component of the point.
```
Comment (alejandro-isaza) [inline PathKit/Sources/PathKit/Point.swift]:
```suggestion
    /// The y-coordinate of the point.
```

**Review by alejandro-isaza — APPROVED** [review]

**Pull request merged**